src/main/java/frc/robot/commands/DriveWithDistance.java

```diff
@@ -0,0 +1,382 @@
+      currentLeftDistance = DriveBase.getInstance().getLeftDistance();
+
+      // The robot is moving forward
+      if (!isTurning) {
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving forward).
+        if (!backWards && currentRightDistance >= desiredRightDistance) {
+          driveBase.tankDrive(0, 0, false);
+          commandFinished = true;
+
+        }
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving backward).
+        else if (backWards && currentRightDistance <= desiredRightDistance) {
+          driveBase.tankDrive(0, 0, false);
+          commandFinished = true;
+        } else {
+          commandFinished = false;
+        }
+
+      }
+      // If the robot is turning, then move the robot for the left side too
+      else {
+
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving forward).
+        if (!rightBackwards && currentRightDistance >= desiredRightDistance) {
+          rightSpeed = 0;
+        }
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving backward).
+        else if (rightBackwards && currentRightDistance <= desiredRightDistance) {
+           rightSpeed = 0;
+          
+        } else {
+          commandFinished = false;
+        }
+
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving forward).
+        if (!leftBackwards && currentLeftDistance >= desiredLeftDistance) {
+          leftSpeed = 0;
+        }
+        // If the robot has reached or surpassed the desired distance, then stop the
+        // robot. Otherwise, keep moving (moving backward).
+        else if (leftBackwards && currentLeftDistance <= desiredLeftDistance) {
+          leftSpeed = 0;
+        } else {
+          commandFinished = false;
+        }
+
+        //If the left and right speed are zero, just stop the robot
+        if(leftSpeed == 0 && rightSpeed == 0){
+          commandFinished = true;
+          end = true;
+        }
+
+        //If the desired and current distance are equal, don't moce the robot
+        if(desiredLeftDistance == currentLeftDistance && desiredRightDistance == currentRightDistance)
+        {
+          commandFinished = true;
+          end = true;
+        }
+
+        //Taking the changed left and right speed and adding their updated values to the drivebase
+        driveBase.tankDrive(leftSpeed, rightSpeed, false);
+
+      }
+    }
+    return commandFinished;
+  }
+
+    // Called once the command ends or is interrupted.
+    @Override
+    public void end(boolean interrupted) {
+        SmartDashboard.putBoolean("Running", false);
```

Comment:
Either remove this or make it more descriptive

Reply:
Removed